### PR TITLE
Minor fix to TEST() - allow positive signed 32-bit immediate operands

### PR DESCRIPTION
**Pull request by noahgibbs**
The third test there is commented out because I can't figure out how to get our assembler and yasm to agree. The (failing) bytes there are what I got from yasm for "TEST qword RAX, dword 0x0FFFFFF7".

With this fix applied, "./miniruby --yjit-call-threshold=1 -e 'a=2'" gets to a panic in branch_stub_hit(), which looks like the one that Alan's latest PR is aimed at.

### CODE REVIEW

File: yjit/src/asm/x86_64/tests.rs [hunk 1]
```diff
@@ -381,6 +381,9 @@ fn test_test() {
     check_bytes("4885c0", |cb| test(cb, RAX, RAX));
     check_bytes("4885f0", |cb| test(cb, RAX, RSI));
     check_bytes("48f74640f7ffffff", |cb| test(cb, mem_opnd(64, RSI, 64), imm_opnd(!0x08)));
+    check_bytes("48f7464008000000", |cb| test(cb, mem_opnd(64, RSI, 64), imm_opnd(0x08)));
+    check_bytes("48f7c108000000", |cb| test(cb, RCX, imm_opnd(0x08)));
+    //check_bytes("48a9f7ffff0f", |cb| test(cb, RAX, imm_opnd(0x0FFFFFF7)));
```
Comment (maximecb) [inline yjit/src/asm/x86_64/tests.rs:386]: Sometimes there's special encodings for RAX. Should try corner cases like `mov RCX, u16::MAX` and `mov RCX, u32::MAX`.

Comment (noahgibbs) [reply yjit/src/asm/x86_64/tests.rs:386]: Ah, you're right, it works fine with RCX but not with RAX. So that'll be that problem at least.

Our assembler doesn't like u32::MAX as an imm_opnd, but that makes sense. It would need to be a large negative number instead. And whenever I supply a uimm_opnd, it wants to compare against ECX rather than RCX (32 bits, not 64.) I can't tell if this is intended behaviour or not...

Anyway, I'll keep adding cases and leave anything commented out that I can't get to pass.

Comment (maximecb) [reply yjit/src/asm/x86_64/tests.rs:386]: Makes sense that u32::MAX doesn't work 👍

**Review by maximecb — APPROVED** [review]

**Review by XrXr — APPROVED** [review]
The third test is using the special `A9 id` opcode for `TEST rax, imm32` which sign extends the immediate.
